<?xml version="1.0"?>

<robot name="quadrotor" xmlns:xacro="http://www.ros.org/wiki/xacro">
    <xacro:arg name="gpu" default="true"/>
    <xacro:property name="gpu" value="$(arg gpu)" />
    <xacro:property name="M_PI" value="3.1415926535897931" />
    
    <!-- Included URDF Files -->
    <xacro:include filename="$(find hector_quadrotor_description)/urdf/quadrotor_base.urdf.xacro" />
    
    <!-- Instantiate quadrotor_base_macro once (has no parameters atm) -->
    <xacro:quadrotor_base_macro />

    <!-- Velodyne HDL-32E mounted upside down below the quadrotor body -->
    <xacro:include filename="$(find velodyne_description)/urdf/HDL-32E.urdf.xacro"/>

    <xacro:HDL-32E parent="base_link" name="velodyne" topic="/velodyne_points" hz="10" samples="500" gpu="${gpu}">
      <origin xyz="0 0 0" rpy="${M_PI} 0 0"/>
    </xacro:HDL-32E>
    
    <!-- Forward facing camera -->
    <xacro:include filename="$(find hector_sensors_description)/urdf/generic_camera.urdf.xacro" />
    <xacro:generic_camera name="front_cam" parent="base_link" ros_topic="camera/image" cam_info_topic="camera/camera_info" update_rate="10" res_x="320" res_y="240" image_format="R8G8B8" hfov="90">
      <origin xyz="0.05 0.0 -0.06" rpy="0 0 0"/>
    </xacro:generic_camera>
    
</robot>